src/devices/cpu/z80/tmpz84c015.h

```diff
@@ -81,6 +81,9 @@ class tmpz84c015_device : public z80_device
 	DECLARE_WRITE_LINE_MEMBER( trg2 ) { m_ctc->trg2(state); }
 	DECLARE_WRITE_LINE_MEMBER( trg3 ) { m_ctc->trg3(state); }
 
+	template <int Channel> void set_clk_trg(u32 clock) { if (m_ctc) m_ctc->set_clk<Channel>(clock); else subdevice<z80ctc_device>(m_ctc.finder_tag())->set_clk<Channel>(clock); }
+	template <int Channel> void set_clk_trg(const XTAL &xtal) { if (m_ctc) m_ctc->set_clk<Channel>(xtal); else subdevice<z80ctc_device>(m_ctc.finder_tag())->set_clk<Channel>(xtal); }
```

Comment:
@smf- can you help with answer as an author?

Reply:
It will depend on whether we emulate hardware that changes the clock speed at run time.

Ideally the finders wouldn't be torn down, so you could still use them during configuration time.